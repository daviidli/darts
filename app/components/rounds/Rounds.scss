.item {
	background-color: rgba(0, 0, 0, 0.4);
	width: 4em;
	height: 1.5em;
	padding: 0.25em;
	display: inline-block;
	text-align: center;
	margin-bottom: 0.5em;
	font-size: large;

	&.first {
		border-radius: 0.25em 0 0 0.25em;
	}

	&.last {
		border-radius: 0 0.25em 0.25em 0;
	}
}
