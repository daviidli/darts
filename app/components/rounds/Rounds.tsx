import React from 'react';
import clsx from 'clsx';
import styles from './Rounds.scss';
import { mappingToString } from '../../utils/dartUtils';
import DartText from '../dartText/DartText';

type Props = {
	round: string[];
};

const Rounds = (props: Props) => {
	const { round } = props;

	return (
		<li>
			{round.map((r, i) => {
				return (
					<span
						key={`${r}-${i}`}
						className={clsx(styles.item, {
							[styles.first]: i === 0,
							[styles.last]: i === round.length - 1
						})}
					>
						<DartText value={mappingToString(r)} />
					</span>
				);
			})}
		</li>
	);
};

export default Rounds;
