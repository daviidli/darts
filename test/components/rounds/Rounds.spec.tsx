import React from 'react';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import Rounds from '../../../app/components/rounds/Rounds';

Enzyme.configure({ adapter: new Adapter() });

const initalProps: string[] = ['1', '20-t', 'b-d'];

describe('Rounds component', () => {
	it('should match snapshot', () => {
		const tree = renderer.create(<Rounds round={initalProps} />).toJSON();
		expect(tree).toMatchSnapshot();
	});
});
