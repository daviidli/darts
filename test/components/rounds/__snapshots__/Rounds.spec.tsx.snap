// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rounds component should match snapshot 1`] = `
<li>
  <span
    className="item first"
  >
    <span
      className=""
    >
      1
    </span>
  </span>
  <span
    className="item"
  >
    <span
      className="triple"
    >
      20
    </span>
  </span>
  <span
    className="item last"
  >
    <span
      className="double"
    >
      Bull
    </span>
  </span>
</li>
`;
